inference-engine/src/mkldnn_plugin/nodes/batch_to_space.cpp

```diff
@@ -17,209 +17,209 @@ namespace Cpu {
 
-            if (batchToSpaceLayer->insData.size() != 4 || batchToSpaceLayer->outData.size() != 1)
-                THROW_IE_EXCEPTION << "'" << batchToSpaceLayer->name << "' layer has incorrect number of input or output edges!";
+            if (batchToSpaceLayer->insData.size() != 4)
+                THROW_IE_EXCEPTION << "'" << batchToSpaceLayer->name << "' has incorrect number of input edges";
 
-            auto inData = batchToSpaceLayer->insData[0].lock();
-            if (inData == nullptr)
-                THROW_IE_EXCEPTION << "'" << batchToSpaceLayer->name << "' layer has nullable input data";
+            if (batchToSpaceLayer->outData.size() != 1)
+                THROW_IE_EXCEPTION << "'" << batchToSpaceLayer->name << "' has incorrect number of output edges";
 
-            if (inData->getLayout() != NCHW && inData->getLayout() != NCDHW)
-                THROW_IE_EXCEPTION << "'" << batchToSpaceLayer->name << "' layer has unsupported layout: " << inData->getLayout();
+            inDims = batchToSpaceLayer->insData[0].lock()->getTensorDesc().getDims();
+            if (inDims.size() < 4)
+                THROW_IE_EXCEPTION << "'" << batchToSpaceLayer->name
+                << "' has incorrect number of input dimensions";
 
-            const auto precision = inData->getTensorDesc().getPrecision();
+            if (inDims.size() > 5)
+                THROW_IE_EXCEPTION << "'" << batchToSpaceLayer->name
+                << "' doesn't support dimensions with rank greater than 5";
+
+            outDims = batchToSpaceLayer->outData[0]->getTensorDesc().getDims();
+            if (inDims.size() != outDims.size())
+                THROW_IE_EXCEPTION << "'" << batchToSpaceLayer->name
+                << "' has incorrect number of input/output dimensions";
+
+            if (inDims[1] != outDims[1])
+                THROW_IE_EXCEPTION << "'" << batchToSpaceLayer->name << "' layer has different IN and OUT channels number";
+
+            const auto precision = batchToSpaceLayer->insData[0].lock()->getTensorDesc().getPrecision();
             const std::set<size_t> supported_precision_sizes = {1, 2, 4, 8};
             if (supported_precision_sizes.find(precision.size()) == supported_precision_sizes.end())
                 THROW_IE_EXCEPTION << "'" << batchToSpaceLayer->name << "' layer has unsupported precision: " << precision.name();
 
-            const SizeVector& in_dims = inData->getTensorDesc().getDims();
-            const SizeVector& out_dims = layer->outData[0]->getTensorDesc().getDims();
-            if (in_dims[1] != out_dims[1])
-                THROW_IE_EXCEPTION << "'" << batchToSpaceLayer->name << "' layer has different IN and OUT channels number";
-
             _block_shape = batchToSpaceLayer->_block_shape;
             _crops_begin = batchToSpaceLayer->_crops_begin;
-            _crops_end = batchToSpaceLayer->_crops_end;
+
+            if (inDims.size() == 4 || inDims.size() == 5) {
```

Comment:
You may use lambda function for create configs to avoid code duplication (as in [pad layer](https://github.com/openvinotoolkit/openvino/blob/f88840d5009a021c648becc23301d4dc9acc8536/inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.cpp#L90) but for `TensorDesc`)

And for example after write:
`createConfig(inDims.size() == 4 ? NHWC : NDHWC);
 createConfig(InferenceEngine::TensorDesc::getLayoutByDims(outDims));`